### PR TITLE
test: Add vitest and some basic integration tests

### PR DESCRIPTION
**Pull request by juliushaertl**
Add vitest as test framework and some rather basic integration tests for the metrics endpoint and socket auth.

```
npm install --save axios
npm install --save-dev vitest
```

### CODE REVIEW

**Comment by juliushaertl** [comment]
🏓  for review ... will resolve the package.json conflict before merge to avoid another conflict.

File: websocket_server/SocketManager.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ export default class SocketManager {
 		this.socketDataManager = new SocketDataManager(storageManager)
 
 		this.io = new SocketIO(server, {
-			transports: ['websocket'],
+			transports: ['websocket', 'polling'],
```
Comment (max-nextcloud) [inline websocket_server/SocketManager.js:26]: Is this related?

**Review by max-nextcloud — APPROVED** [review]
lgtm. Just one minor question.

File: websocket_server/SocketManager.js (hunk 1 above)
Comment (juliushaertl) [reply websocket_server/SocketManager.js:26]: Yes, it is actually a fallback as vitest was failing on the websocket handling (didn't dig too deep into that, but polling is the regular fallback mechanism and would also be needed for #119)
